src/chat/delete_chat_modal.rs

```diff
@@ -0,0 +1,217 @@
+                            border_width: 1.2,
+                            color: #fff,
+                        }
+
+                        text: "Cancel"
+                        draw_text:{
+                            text_style: <REGULAR_FONT>{font_size: 10},
+                            color: #x0
+                        }
+                    }
+
+                    delete_button = <MoxinButton> {
+                        width: Fit,
+                        height: Fit,
+                        padding: {top: 10, bottom: 10, left: 14, right: 14}
+
+                        draw_bg: {
+                            instance radius: 2.0,
+                            color: #D92D20,
+                        }
+
+                        text: "Delete"
+                        draw_text:{
+                            text_style: <REGULAR_FONT>{font_size: 10},
+                            color: #fff
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[derive(Clone, DefaultNone, Debug)]
+pub enum DeleteChatAction {
+    ChatSelected(ChatID),
+    None,
+}
+
+#[derive(Live, LiveHook, Widget)]
+pub struct DeleteChatModal {
+    #[deref]
+    view: View,
+    #[rust]
+    chat_id: ChatID,
+}
+
+impl Widget for DeleteChatModal {
+    fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
+        self.view.handle_event(cx, event, scope);
+        self.widget_match_event(cx, event, scope);
+    }
+
+    fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, walk: Walk) -> DrawStep {
+        let chat_title = scope
+            .data
+            .get::<Store>()
+            .unwrap()
+            .chats
+            .saved_chats
+            .iter()
+            .map(|chat| chat.borrow())
+            .find(|chat| chat.id == self.chat_id)
+            .unwrap()
+            .get_title()
+            .to_string();
+
+        let prompt_text = format!(
+            "Are you sure you want to delete {}?\nThis action cannot be undone.",
+            chat_title
+        );
+        self.label(id!(wrapper.body.delete_prompt))
+            .set_text(&prompt_text);
+
+        self.view
+            .draw_walk(cx, scope, walk.with_abs_pos(DVec2 { x: 0., y: 0. }))
+    }
+}
+
+impl WidgetMatchEvent for DeleteChatModal {
+    fn handle_actions(&mut self, cx: &mut Cx, actions: &Actions, scope: &mut Scope) {
+        let widget_uid = self.widget_uid();
+
+        if self.button(id!(close_button)).clicked(actions) {
+            cx.widget_action(widget_uid, &scope.path, PortalAction::Close);
+        }
+
+        if self
+            .button(id!(wrapper.body.actions.delete_button))
+            .clicked(actions)
+        {
+            let store = scope.data.get_mut::<Store>().unwrap();
+            store.delete_chat(self.chat_id);
+            cx.widget_action(widget_uid, &scope.path, PortalAction::Close);
+            cx.redraw_all();
```

Comment:
Tell makepad to fully redraw to incorporate the changes from the store.

Reply:
I don't love this, but I get that you're avoiding the creation of some kind of action to listen to and redraw in the relevant parts of the UI. 
I think is fine for now, eventually we could add metadata to the `PortalAction::Close` action (so we listen to the right one) and redraw based on that.